ibis/backends/base/sql/compiler/base.py

```diff
@@ -116,7 +116,5 @@ def compile(self):
 def _extract_common_table_expressions(nodes):
     # filter out None values
     nodes = list(filter(None, nodes))
-    counts = an.find_subqueries(nodes)
-    duplicates = [op for op, count in counts.items() if count > 1]
-    duplicates.reverse()
-    return duplicates
+    subqueries = an.find_subqueries(nodes)
+    return tuple(reversed(subqueries))
```

Comment:
Not a blocker, but it seems a bit wasteful to return a `tuple` from `find_subqueries`, reverse it, and then turn that into a tuple, when we can return a `list` and reverse it in place.